src/storage_persister.rs

```diff
@@ -60,38 +77,122 @@ impl StoragePersister {
     {
-        /*let mut result = Vec::new();
-        // TODO: Handle unwrap().
+        // Get local ChannelMonitors
+        let mut local_channel_monitors =
+            match self.fs_persister.read_channelmonitors(&*keys_manager) {
+                Ok(v) => v,
+                Err(e) => {
+                    return Err(e)
+                        .map_to_permanent_failure("Failed to read channel monitors from disk")
+                }
+            };
+
+        // Get remote ChannelMonitors
+        let mut remote_channel_monitors = Vec::new();
         for key in self
             .storage
             .list_objects(MONITORS_BUCKET.to_string())
-            .unwrap()
+            .map_to_runtime_error(
+                RuntimeErrorCode::RemoteStorageServiceUnavailable,
+                "Failed to get list of ChannelMonitors from remote storage",
+            )?
         {
-            // TODO: Handle unwrap().
             let data = self
                 .storage
                 .get_object(MONITORS_BUCKET.to_string(), key.clone())
-                .unwrap();
+                .map_to_runtime_error(
+                    RuntimeErrorCode::RemoteStorageServiceUnavailable,
+                    format!("Failed to get ChannelMonitor {} from remote storage", key),
+                )?;
             let mut buffer = Cursor::new(&data);
             match <(BlockHash, ChannelMonitor<Signer>)>::read(&mut buffer, &*keys_manager) {
                 Ok((blockhash, channel_monitor)) => {
                     debug!(
-                        "Successfully read ChannelMonitor {} from storage",
+                        "Successfully read ChannelMonitor {} from remote storage",
                         channel_monitor.get_funding_txo().0.to_channel_id().to_hex()
                     );
-                    result.push((blockhash, channel_monitor));
+                    remote_channel_monitors.push((blockhash, channel_monitor));
                 }
                 Err(e) => {
-                    error!("Failed to deserialize ChannelMonitor `{}`: {}", key, e);
+                    error!(
+                        "Failed to deserialize remote ChannelMonitor `{}`: {}",
+                        key, e
+                    );
                     // TODO: Should we return this information to the caller?
+                    //      A corrupt remote ChannelMonitor could be harmless if in this case we
+                    //      load the local ChannelMonitors and the situation gets fixed. If this
+                    //      is a wallet recovery, we will need to load the remote ChannelMonitors,
+                    //      and this channel will be lost.
+                }
+            }
+        }
+
+        if local_channel_monitors.is_empty() {
+            // If we don't have any local ChannelMonitors, but have 1 or more remote ChannelMonitors,
+            // we can assume that this is a new app installation and we want to use the remote state.
```

Comment:
Maybe we could ask the consumer to tell us in `Config` if it's a recovery start or not? Otherwise, we can only check by trying to get the remote `ChannelMonitors`. I refactored a bit to make it clear that we learn in `read_channel_monitors()` if we are in a normal, fresh, or recovery start. 

The naming for `StartupVariant` could be improved maybe. I'm open to ideas.